.vsts-ci/ci.yml

```diff
@@ -0,0 +1,26 @@
+name: PR-$(System.PullRequest.PullRequestNumber)-$(Date:yyyyMMdd)$(Rev:.rr)
+resources:
+- repo: self
+  clean: true
+
+phases:
+
+- phase: Set_Build_Name
+#  displayName: Set Build Name
+  condition: ne(variables['Build.Reason'], 'PullRequest')
+  queue:
+    name: Hosted Ubuntu 1604
+  steps:
+  - powershell:  Write-Host "##vso[build.updatebuildnumber]$env:BUILD_SOURCEBRANCHNAME-$env:BUILD_SOURCEVERSION-$((get-date).ToString("yyyyMMddhhss"))"
+    displayName: Set Build Name for Non-PR
+    condition: ne(variables['Build.Reason'], 'PullRequest')
+
+- template: ./phase.yml
+  parameters:
+    SnapChannel: stable
+    SnapName: powershell
+
+- template: ./phase.yml
+  parameters:
+    SnapChannel: preview
```

Comment:
I thought we were publishing the `powershell-preview` package to its `stable` channel?

Reply:
It seems like this is used for something else... just a switch statement. Is it needed? Couldn't we just use the SnapName?

Reply:
very few characters are allowed in the phase name, but it also must be unique.

Reply:
also, there is no switch statement in the language.  it would end up being a bunch of if statement.

Reply:
But we can use just the Channel... I updated the code